extension/entitystore/extension.go

```diff
@@ -48,6 +48,7 @@ type EntityStore struct {
 	logger *zap.Logger
 	config *Config
 	done   chan struct{}
+	ready  bool
```

Comment:
If you don't want to use a lock, you could use an atomic.Bool (https://pkg.go.dev/sync/atomic#Bool)